minecraft-server/start.sh

```diff
@@ -11,5 +11,4 @@ while lsof -- /start-minecraft; do
   echo -n "."
   sleep 1
 done
-echo "...switching to user 'minecraft'"
-exec sudo -E -u minecraft /start-minecraft
```

Comment:
The `sudo` is needed to avoid files in the attached volume getting created as root. I'm thinking you ran into issues with `/root` since `$HOME` might have still been referring to root's. Adding an

    export HOME=/data

to the top of `start-minecraft.sh` might resolve that.